src/models/db/users.js

```diff
@@ -1,27 +1,52 @@
-const db = require('./db')
+const bcrypt = require('bcrypt');
+const db = require('./db');
 
-const signUp = (user) => {
+const encryptPassword = (password) => {
+  const saltRounds = 10;
+  return bcrypt.hash(password, saltRounds);
+};
+
+const create = (user) => {
   return db.one(`
     INSERT INTO
-      users(first_name, email, encrypted_password)
+      users(full_name, email, password)
     VALUES
       ($1, $2, $3)
     RETURNING
       *
-  `, [user.first_name, user.email, user.password])
+  `, [user.full_name, user.email, user.password])
   .catch(console.log)
 };
 
-const signIn = (email, password) => {
+const signUp = (user) => {
+    return encryptPassword(user.password).then((hash) => {
+      user.password = hash
+      return create(user);
+    }).catch(console.log)
+};
+
+const findByEmail = (email) => {
   return db.oneOrNone(`
     SELECT * FROM
       users
     WHERE
       email=$1
-      password=$2
-  `, [email, password])
+  `, [email])
   .catch(console.log)
-}
+};
+
+const signIn = (email, password) => {
+  return findByEmail(email)
+  .then((user) => {
+    return bcrypt.compare(password, user.password)
+    .then((validUser) => {
+      if (!validUser) {
+        throw new Error ('Invalid Username/Password')
+      }
```

Comment:
do we need to return validUser here? I imagine it's working or it wouldn't be in your PR...?

Reply:
Complete 👍